lib/lists.nix

```diff
@@ -1330,35 +1330,47 @@ rec {
           pairs);
 
   /**
-    Compare two lists element-by-element.
+    Compare two lists element-by-element with a comparison function `cmp`.
+
+    List elements are compared in order, the first non-equal pair of elements determines the result.
```

Comment:
Let's be explicit

```suggestion
    List elements are compared in order, the first non-equal pair of elements determines the result, which is expressed as an integer:
    - `-1` if the first element of the first unequal pair is less
    - `1` if the first element of the first unequal pair is greater
    - `0` if all elements are equal
```

Reply:
This is already described in the `comparison function` argument. I would try to not duplicate this information.

Reply:
The output of the function should be at the top, and it's not strictly implied by the signature of the comparison function. We can also describe the output indirectly, e.g.

```
The result is the integer returned by the comparison function for the first unequal pair in the list, or `0`.
```